File: Kombinatorika/Faktorial-Permutasi-Kombinasi.c
#include <stdio.h>

void faktorial ();
void permutasi ();
void kombinasi ();

int main ()
{
    unsigned short pilihan = 0;

    do
    {
        puts ("========== KOMBINATORIKA ==========");
        puts ("1. Faktorial");
        puts ("2. Permutasi");
        puts ("3. Kombinasi");
        puts ("4. Keluar");
        puts ("===================================");
        printf ("Pilih salah satu: ");
        scanf ("%hu", &pilihan);

        fflush (stdin);

        switch (pilihan)
        {
        case 1: faktorial ();
                break;
        case 2: permutasi ();
                break;
        case 3: kombinasi();
                break;
        case 4: puts ("Terima kasih telah menggunakan program ini!");
                break;
        default: puts ("Input error");
                break;
        }
    } while (pilihan != 4);

    return 0;
}

void faktorial ()
{
    int angka = 0, hasil = 1;

    printf ("Masukkan bilangan faktorial: ");
    scanf ("%i", &angka);

    for (int i = angka; i >= 1; i--)
    {
        hasil *= i;
    }

    printf ("%i! = %i\n", angka, hasil);
}

void permutasi ()
{
    int n = 0, p = 1, k = 0;

    printf ("Masukkan bilangan keseluruhan: ");
    scanf ("%i", &n);
    printf ("Masukkan bilangan asli (tidak termasuk duplikat): ");
    scanf ("%i", &k);

    for (int i = n; i > n - k; i--)
    {
        p *= i;
    }

    printf ("Hasil: %i\n", p);
}

void kombinasi ()
{
    int n = 0, r = 0, n_faktorial = 1, r_faktorial = 1, nr_faktorial = 1;

    printf ("Masukkan bilangan keseluruhan: ");
    scanf ("%i", &n);
    printf ("Masukkan bilangan pilihan: ");
    scanf ("%i", &r);

    for (int i = n; i > 0; i--)
    {
        n_faktorial *= i;
    }
    for (int i = n - r; i > 0; i--)
    {
        nr_faktorial *= i;
    }
    for (int i = r; i > 0; i--)
    {
        r_faktorial *= i;
    }

    printf ("Hasil: %i\n", n_faktorial / (r_faktorial * nr_faktorial));
}

File: Kombinatorika/Faktorial-Permutasi-Kombinasi.py
def main ():
    while True:
        print ("========== KOMBINATORIKA ==========")
        print ("1. Faktorial")
        print ("2. Permutasi")
        print ("3. Kombinasi")
        print ("4. Keluar")
        print ("===================================")
        pilihan = int (input ("Pilih salah satu: "))
        if pilihan == 1:
            faktorial ()
        elif pilihan == 2:
            permutasi ()
        elif pilihan == 3:
            kombinasi ()
        elif pilihan == 4:
            print ("Terima kasih telah menggunakan program ini!")
            break
        else:
            print ("Input error")

def faktorial ():
    angka = int (input ("Masukkan bilangan faktorial: "))
    hasil = 1
    for i in range (angka, 0 ,-1):
        hasil *= i
    print (f"{angka}! = {hasil}")

def permutasi ():
    n = int (input ("Masukkan bilangan keseluruhan: "))
    k = int (input ("Masukkan bilangan asli (tidak termasuk duplikat): "))
    p = 1
    for i in range (n, n - k, -1):
        p *= i
    print (f"Hasil: {p}")

def kombinasi ():
    n = int (input ("Masukkan bilangan keseluruhan: "))
    r = int (input ("Masukkan bilangan asli (tidak termasuk duplikat): "))
    n_faktorial = nr_faktorial = r_faktorial = 1
    for i in range (n, 0, -1):
        n_faktorial *= i
    for i in range (n - r, 0, -1):
        nr_faktorial *= i
    for i in range (r, 0, -1):
        r_faktorial *= i
    print (f"Hasil: {n_faktorial / (r_faktorial * nr_faktorial)}")

if __name__ == "__main__":
    main ()